src/libpriv/rpmostree-core.c

```diff
@@ -753,6 +753,12 @@ rpmostree_context_setup (RpmOstreeContext    *self,
   /* This is what we use as default. */
   dnf_context_set_rpm_macro (self->dnfctx, "_dbpath", "/" RPMOSTREE_RPMDB_LOCATION);
 
+  // Default for now for compat reasons
+  const char *db_backend = "bdb";
+  if (self->treefile_rs && ror_treefile_get_sqlite (self->treefile_rs))
+    db_backend = "sqlite";
```

Comment:
So then this would be

```suggestion
  const char *db_backend = "bdb";
  if (self->treefile_rs)
    db_backend = ror_treefile_get_rpmdb (self->treefile_rs) ?: db_backend;
```